File: server/models/quiz.go
package models

import "gorm.io/gorm"

type Quiz struct {
    gorm.Model
    Title string `json:"title"`
    Questions []Question `json:"questions"`
    CourseId string `json:"course_id"`
}

type Question struct {
    gorm.Model
    Title string `json:"title"`
    QuizID uint   `json:"quiz_id"`
    Answers []Answer `json:"answers"`
}

type Answer struct {
    gorm.Model
    Title string `json:"title"`
    IsCorrect bool `json:"is_correct"`
    QuestionID uint `json:"question_id"`
}

func (Quiz) TableName() string {
    return "quizzes" // Specify the correct table name
}

File: client/src/types/Quiz.ts
export type QuizAnswers ={
    ID?:number
    title:string,
    is_correct:boolean
    question_id?:number
}
export type QuizQuestions = {
    ID?:number
    title:string,
    answers?:QuizAnswers[]
}

export type Quiz = {
    ID?:number
    title:string,
    course_id?:number
    questions?: QuizQuestions[]
}


type SelectedAnswers ={
    quiz_id:number,
    question_id:number,
    answer_id:number,
    points:number,


}

export type SubmittedQuizData ={
    course_id:number,
    professor_id:number,
    student_id:number,
    grade:number,
    selected_answers?:SelectedAnswers[]

}